src/NativeScript/GlobalObject.mm

```diff
@@ -118,6 +119,27 @@ static void microtaskRunLoopSourcePerformWork(void* context) {
     self->drainMicrotasks();
 }
 
+static void runLoopBeforeWaitingPerformWork(CFRunLoopObserverRef observer, CFRunLoopActivity activity, void* info) {
+    GlobalObject* self = static_cast<GlobalObject*>(info);
+    VMEntryScope* currentEntryScope = self->vm().entryScope;
+    if (self->vm().topCallFrame && currentEntryScope && !currentEntryScope->didPopListeners().isEmpty()) {
+        FFIFunctionCall* function_call = static_cast<FFIFunctionCall*>(self->vm().topCallFrame->callee());
```

Comment:
I think that you can use `jsDynamicCast()` here. That will make sure that the function call is indeed a `FFIFunctionCall` or return null otherwise so that it can be checked below.